download-metadata-and-files.tf

```diff
@@ -0,0 +1,56 @@
+locals {
+  disaster_recovery_bucket_name = "${local.environment}-disaster-recovery"
+  lambda_name                   = "${local.environment}-download-files-metadata"
+  queue_name                    = "${local.environment}-download-files-metadata"
+}
+
+module "download_metadata_and_files_lambda" {
+  source        = "git::https://github.com/nationalarchives/da-terraform-modules//lambda?ref=add-nat-instance-and-vpc-policy"
+  function_name = local.lambda_name
+  handler       = "uk.gov.nationalarchives.Lambda::handleRequest"
+  policies = {
+    "${local.lambda_name}-policy" = templatefile("./templates/iam_policy/download_files_metadata_policy.json.tpl", {
+      secrets_manager_secret_arn   = "arn:aws:secretsmanager:eu-west-2:${var.dp_account_number}:secret:sandbox-preservica-6-preservicav6login-INFTcQ",
+      download_files_sqs_queue_arn = module.download_files_sqs.sqs_arn
+      disaster_recovery_bucket     = module.disaster_recovery_bucket
+      bucket_name                  = local.disaster_recovery_bucket_name
+      account_id                   = var.dp_account_number
+      lambda_name                  = local.lambda_name
+    })
+  }
+  memory_size = 512
+  runtime     = "java17"
+  vpc_config = {
+    subnet_ids         = module.vpc.private_subnets
+    security_group_ids = [module.outbound_https_access_only.security_group_id]
+  }
+  plaintext_env_vars = {
+    DR_BUCKET_NAME  = local.disaster_recovery_bucket_name
+    API_SECRET_NAME = aws_secretsmanager_secret.preservica_secret.name
+    PRESERVICA_URL  = data.aws_ssm_parameter.preservica_url.value
+  }
+  tags = {
+    Name      = local.lambda_name
+    CreatedBy = "dp-terraform-environments"
+  }
+}
+
+module "download_files_sqs" {
+  source                   = "git::https://github.com/nationalarchives/da-terraform-modules//sqs"
+  queue_name               = local.queue_name
+  sqs_policy               = templatefile("./templates/sqs/sqs_access_policy.json.tpl", { role_arn = module.download_metadata_and_files_lambda.lambda_role_arn, account_id = var.dp_account_number, queue_name = local.queue_name })
```

Comment:
Why does this lambda need to send to SQS?

Reply:
It doesn't need to send to SQS. This is supposed to be the queue that triggers the lambda. I've got it the wrong way round. I've changed it now so it is the queue that triggers the lambda.